File: bubble_sort.ts
function bubbleSort(arr: (number | string)[]) {
  const timesToIterateArray = arr.length - 1
  let temp: string | number
  let swapped: boolean

  for (let i = 0; i < timesToIterateArray; i++) {
    swapped = false
    for (let j = 0; j < timesToIterateArray - i; j++) {
      const value = arr[j]
      const nextValue = arr[j + 1]
      if (value > nextValue) {
        temp = value
        arr[j] = nextValue
        arr[j + 1] = temp
        swapped = true
      }
    }
    if (!swapped) break
  }
  return arr
}

const arr = [5, 3, 8, 4, 2, 1, 9, 6, 7]
console.log(bubbleSort(arr))

File: bubble_sort.py
def bubble_sort(arr):
    times_to_iterate_array = len(arr) - 1
    temp = 0
    swapped = False

    for i in range(0, times_to_iterate_array): 
        swapped = False
        for j in range(0,times_to_iterate_array - i):
            value = arr[j]
            nextValue = arr[j + 1]
            if (value > nextValue): 
                temp = value
                arr[j] = nextValue
                arr[j+1] = temp
                swapped = True
        if swapped == False: 
            break
    return arr

arr = [5, 3, 8, 4, 2, 1, 9, 6, 7]
print(bubble_sort(arr))